src/status/status.factory.js

```diff
@@ -0,0 +1,209 @@
+const { stripIndent } = require('common-tags');
+
+
+module.exports = (logProvider, stateProvider, logger) => {
+    //Helper functions
+    function activeMembers(events) {
+        for(var i=0; i < event.length; i++){
+            var obj = events.logs[i];
+                if(obj.action == "Participated"){
+                    return ({obj})
+                }
+            }
+    }
+
+    function activeEvents(events){
+     for(var i=0; i < events.team_id.length; i++){
+         var obj = events;
+         if(obj.halted == true){
+                return ({obj});
+            }
+        }
+    }
+
+    function happenedInPeriodUser(date, action, back){
+        const alerts = ({
+            notified:0,
+            missed:0
+        });
+        const settingDate = new Date();
+        settingDate.setDate(date);
+        if( back != 'all')
+        {
+            const hasHappened = settingDate - back;
+            while(hasHappened.getUTCDate() < settingDate.getUTCDate()){
+                if(action == "Participated") { alerts.notified =+ 1;}
+                else { alerts.missed =+ 1;}
+            }
+        }
+        else
+        {
+            const hasHappened = settingDate - getFirstEvent();
+            while(hasHappened.getUTCDate < settingDate.getUTCDate()){
+                if(action == "Participated") { alerts.notified =+ 1;}
+                else { alerts.missed =+ 1;}
+            }
+        }
+        return alerts;
+    }
+    function happenedInPeriodEvent(date, events, back){
+        const alerts = {};
+
+        const settingDate = new Date();
+        settingDate.setDate(date);
+        if( back != 'all')
+        {
+            const hasHappened = settingDate - back;
+            while(hasHappened.getUTCDate() < settingDate.getUTCDate()){
+                alert =+ 1;
+            }
+        }
+        else
+        {
+            const hasHappened = settingDate - getFirstEvent();
+            while(hasHappened.getUTCDate < settingDate.getUTCDate()){
+                alert =+ 1;
+            }
+        }
+        return alerts;
+    }
+
+    //Returns user status overall
+    function userLevelStatus(logs, events) {
+        const userStatus = new Map(activeMembers(events).map(user_id => ([user_id, {
+            lastWeek: {notified: 0, missed: 0},
+            lastMonth: {notified: 0, missed: 0},
+            all: {notified: 0, missed: 0}
+        }])))
+
+        logs.filter(({user_id}) => userStatus.has(user_id)).forEach(({user_id, date, action}) => {
+            const wasInLastWeek = happenedInPeriodUser(date, action, 7);
+            const wasInLastMonth = happenedInPeriodUser(date, action, 30);
+            const wasInAll = happenedInPeriodUser(date, action, 'all'); 
+            const prevStatus = userStatus.get(user_id);
+
+            const nextStatus = ({
```

Comment:
you're overriding the `nextStatus` in every iteration and lose everything you calculated in the last iteration. In this way, you would only have a two values 0, 1.